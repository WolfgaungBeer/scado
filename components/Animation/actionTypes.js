export const ANIMATE = '@SCADO/ANIMATION/ANIMATE';
export const REMOVE = '@SCADO/ANIMATION/REMOVE';
