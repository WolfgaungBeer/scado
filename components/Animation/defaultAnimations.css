@keyframes rotateAnimation {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}

.rotateAnimation {
    animation-name: rotateAnimation;
    animation-duration: 2s;
}

@keyframes slideInAnimation {
    from {
        transform: translateX(100px);
    }

    to {
        transform: translateX(0px);
    }
}

.slideInAnimation {
    animation-name: slideInAnimation;
    animation-duration: 1s;
}

@keyframes slideOutAnimation {
    from {
        transform: translateX(0px);
    }

    to {
        transform: translateX(100px);
    }
}

.slideOutAnimation {
    animation-name: slideOutAnimation;
    animation-duration: 1s;
}

@keyframes enterAnimation {
    from {
        opacity: 0;
    }

    to {
        opacity: 1;
    }
}

.enterAnimation {
    animation-name: enterAnimation;
    animation-duration: 1s;
}

@keyframes leaveAnimation {
    from {
        opacity: 1;
    }

    to {
        opacity: 0;
    }
}

.leaveAnimation {
    animation-name: leaveAnimation;
    animation-duration: 1s;
}
