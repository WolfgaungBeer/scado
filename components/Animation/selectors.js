const animationData = (state, id) => state.scado.animations[id] || {};

export default animationData;
