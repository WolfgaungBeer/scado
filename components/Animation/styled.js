import styled from 'styled-components';

export const AnimationWrapper = styled.div`
    display: block;
`;

AnimationWrapper.displayName = 'AnimationWrapper';
