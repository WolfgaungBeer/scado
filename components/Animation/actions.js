import * as types from './actionTypes';

export const animate = (payload) => {
    return { type: types.ANIMATE, payload };
};

export const setClassName = (payload) => {
    return { type: types.SET_CLASSNAME, payload };
};
