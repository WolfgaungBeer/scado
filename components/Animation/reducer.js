import { ANIMATE, REMOVE } from './actionTypes';
import initialState from './initialState';

const reducer = (state = initialState, action) => {
    switch (action.type) {
    case ANIMATE: {
        const newState = { ...state };
        const { id, type, className } = action.payload;
        newState[id] = { type, className };
        return newState;
    }
    case REMOVE: {
        const newState = { ...state };
        const id = action.payload;
        delete newState[id];
        return newState;
    }
    default: {
        return state;
    }
    }
};

export default reducer;
