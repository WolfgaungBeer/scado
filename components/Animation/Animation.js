import React, { PureComponent } from 'react';
import { string, bool, object, func, node } from 'prop-types';
import { AnimationWrapper } from './styled';

const propTypes = {
    id: string.isRequired,
    hideOnMount: bool,
    animationData: object.isRequired,
    remove: func.isRequired,
    children: node
};

const defaultProps = {
    hideOnMount: undefined,
    children: undefined,
};

class Animation extends PureComponent {

    constructor(props) {
        super(props);
        this.state = { visible: !props.hideOnMount };
    }

    componentWillReceiveProps(nextProps) {
        const { animationData, remove } = nextProps;
        const { visible } = this.state;
        const { type } = animationData;
        const div = this.animationContainer;

        if (!div && type && type === 'enter') {
            this.setState({ visible: true });
        }

        if (div && type && type === 'leave') {
            div.addEventListener('animationend', () => {
                remove();
                this.setState({ visible: false });
            }, false);
        }
    }

    render() {
        const { hideOnMount, animationData, children } = this.props;
        const { visible } = this.state;
        const { className } = animationData;

        if (!visible) {
            return (null);
        }

        return (
            <AnimationWrapper innerRef={(div) => { this.animationContainer = div; }} className={className}>
                {children}
            </AnimationWrapper>
        );
    }

}

Animation.propTypes = propTypes;
Animation.defaultProps = defaultProps;

export default Animation;
