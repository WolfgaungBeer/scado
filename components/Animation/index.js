import { connect } from 'react-redux';
import { animationData } from './selectors';
import animationActions from './actions';
import Animation from './Animation';
import './defaultAnimations.css';

const mapStateToProps = (state, ownProps) => ({
    animationData: animationData(state, ownProps.id),
});

const mapDispatchToProps = (dispatch, ownProps) => ({
    remove: () => dispatch(animationActions.remove(ownProps.id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Animation);
