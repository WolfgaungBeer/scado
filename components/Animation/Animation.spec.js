import React from 'react';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Animation from './';

it('renders the Animation Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Animation hideOnMount inline><div>test</div></Animation>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Animation id="test1" inline><div>test</div></Animation>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();

    const tree3 = renderer.create(
        <Wrapper>
            <Animation id="test2"><div>test</div></Animation>
        </Wrapper>,
    ).toJSON();
    expect(tree3).toMatchSnapshot();
});
