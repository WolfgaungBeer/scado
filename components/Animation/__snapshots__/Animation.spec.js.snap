// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Animation Component correctly 1`] = `null`;

exports[`renders the Animation Component correctly 2`] = `
<div
  className="testClass1 sc-bdVaJa hxOzgB"
>
  <div>
    test
  </div>
</div>
`;

exports[`renders the Animation Component correctly 3`] = `
<div
  className="testClass2 sc-bdVaJa iBvyVz"
>
  <div>
    test
  </div>
</div>
`;
